app/javascript/components/public_health/CustomExport.js

```diff
@@ -0,0 +1,508 @@
+                            <span>
+                              {filter.relativeOption === 'custom'
+                                ? `in the ${filter.value?.when} ${filter.value?.number} ${filter.value?.unit}`
+                                : filter.relativeOption}
+                            </span>
+                          )}
+                          {filter.filterOption?.type === 'date' && (
+                            <span>
+                              {filter.dateOption}{' '}
+                              {filter.dateOption === 'within'
+                                ? moment(filter.value?.start).format('MM/DD/YYYY') + ' and ' + moment(filter.value?.end).format('MM/DD/YYYY')
+                                : moment(filter.value).format('MM/DD/YYYY')}
+                            </span>
+                          )}
+                          {!['boolean', 'relative', 'date'].includes(filter.filterOption?.type) && <span>{filter.value}</span>}
+                        </Badge>
+                      );
+                    })}
+                  </div>
+                )}
+                <Form.Check
+                  id="select-monitoree-records-all"
+                  type="radio"
+                  className="px-1"
+                  label={<a onClick={() => this.handleSelectedRecordsChange('all')}>All monitoree records ({this.props.all_monitorees_count})</a>}
+                  onChange={() => this.handleSelectedRecordsChange('all')}
+                  checked={this.state.selected_records === 'all'}
+                />
+                <Form.Check
+                  id="select-monitoree-records-custom"
+                  type="radio"
+                  className="px-1 mb-2"
+                  label={
+                    <a onClick={() => this.handleSelectedRecordsChange('custom')}>
+                      Only include monitoree records that meet the following criteria ({this.state.filtered_monitorees_count}):
+                    </a>
+                  }
+                  onChange={() => this.handleSelectedRecordsChange('custom')}
+                  checked={this.state.selected_records === 'custom'}
+                />
+                {this.state.selected_records === 'custom' && (
+                  <PatientsFilters
+                    authenticity_token={this.props.authenticity_token}
+                    jurisdiction_paths={this.props.jurisdiction_paths}
+                    jurisdiction={this.props.jurisdiction}
+                    query={this.state.custom_patient_query}
+                    onQueryChange={(field, value, cb) =>
+                      this.setState(
+                        state => {
+                          return {
+                            custom_patient_query: { ...state.custom_patient_query, [field]: value },
+                          };
+                        },
+                        () => {
+                          this.getPatientCount();
```

Comment:
I want to make sure that we're all aware that whenever the user changes something in their custom filters, we're making a full line-list query for the count. I understand why we are doing this, but it is more so just something I want to point out for awareness to other reviewers - @holmesie. We already take hits from all the tab counts so it's on my mind.

Another thought on this - when testing this locally and on demo it's going to work really fast bcause there's a smaller number of monitorees. In cases where folks have "all" selected for everything and we're talking 10s of thousands of monitorees (or more) on prod, should we consider having a loading circle here or something if the count takes a few seconds?

Reply:
yeah, having some kind of loading indicator would not be a bad idea

Reply:
https://github.com/SaraAlert/SaraAlert/pull/562/commits/1f13143f8968feff3c7e9ada6139b47e5cd4ecff#diff-14793b3bc71143b45317a3d08474d370db9ffdfb9643e754341eeda1ab5f096bR261

Reply:
Good idea!